zebra/interface.c

```diff
@@ -769,8 +769,20 @@ void if_nbr_ipv6ll_to_ipv4ll_neigh_update(struct interface *ifp,
 
 	inet_pton(AF_INET, buf, &ipv4_ll);
 
-	ipv6_ll_address_to_mac(address, (u_char *)mac);
-	kernel_neigh_update(add, ifp->ifindex, ipv4_ll.s_addr, mac, 6);
+  	ipv6_ll_address_to_mac(address, (u_char *)mac);
+  
+	/*
+	* Remove existed arp record for the interface, 
+	* because netlink protocol does not support message for update. 
+	* supported commands RTM_NEWNEIGH or RTM_DELNEIGH
+	*/
+
+	kernel_neigh_update (0, ifp->ifindex, ipv4_ll.s_addr, mac, 6);
+  
+  	/*
+   	*  Add arp record
```

Comment:
* Please use kernel style for multiline comments
* This could be a single line comment
* Please use tabs instead of spaces